src/main/webapp/scripts/region-selector.js

```diff
@@ -0,0 +1,17 @@
+function getRegions() {
```

Comment:
You are currently adding "dead code" (code that won't run). When adding functionality, you should also add its usage (unless you creating an API method).  In this case, it is showing the list of regions in the UI. Please add this. Also add test for that.